tests/apollo/util/test_base.py

```diff
@@ -51,22 +51,26 @@ async def wrapper(*args, **kwargs):
 
     return decorator
 
-def retry_test(max_retries: int):
+def repeat_test(max_repeats: int, break_on_first_failure: bool, break_on_first_success: bool):
     """
     Runs a test multiple times until a run succeeds
     """
-
+    assert not (break_on_first_failure and break_on_first_success), \
+        "both flags break_on_first_failure and break_on_first_success cannot be enable at the same time!"
     def decorator(async_fn):
         @wraps(async_fn)
         async def wrapper(*args, **kwargs):
-            for i in range(1, max_retries + 1):
+            for i in range(1, max_repeats + 1):
                 try:
+                    print(f"Running iteration {i}/{max_repeats}, (break_on_first_failure={break_on_first_failure},"
```

Comment:
Log instead of printing, but I would not print this at all (just on failure)

Reply:
The main use if it will be done from the command line on the make file when someone wants to run a test 100 times - they want to see this printing to understand how many runs are left.
Since we embed the repeats in tests, it makes sense to print here and not to logs, as looking at the logs is not user-friendly - no one who chooses to run a test case 100 times goes to look at the logs when they want to know how many tests are left + there I might be issued with flashing which I don't want to deal with. 

Bottom line - I think it is better to leave it, but with one correction - print it only when num_repeats > 1.